core/core.go

```diff
@@ -421,11 +426,32 @@ func (app *App) Start(ctx context.Context) error {
 		orderWatcherErrChan <- app.orderWatcher.Watch(innerCtx)
 	}()
 
-	// Backfill block events if needed. This is a blocking call so we won't
-	// continue set up until its finished.
-	if err := app.blockWatcher.BackfillEventsIfNeeded(innerCtx); err != nil {
+	latestBlockProcessed, err := app.blockWatcher.GetLatestBlockProcessed()
+	if err != nil {
 		return err
 	}
+	blocksElapsed := big.NewInt(0)
+	// If has processed blocks before
+	if latestBlockProcessed != nil {
+		latestBlockProcessedNumber := latestBlockProcessed.Number
+		latestBlock, err := app.blockWatcher.GetLatestBlock()
+		if err != nil {
+			return err
+		}
+		latestBlockNumber := latestBlock.Number
+		blocksElapsed = big.NewInt(0).Sub(latestBlockNumber, latestBlockProcessedNumber)
```

Comment:
When profiling performance in the browser, methods in the `math/big` package show up often in the list of those responsible for high CPU usage. Generally, we need to go through and optimize our usage of `math/big` and only use it when necessary. This is a case where I don't think it's necessary. I understand that the Ethereum RPC interface gives us the current block number as a `big.Int` and that's fine. Can we still convert to `int64` as early as possible and then do the math on `int64` types?

Reply:
Agreed. Made this change here and will keep this in mind in future work.